src/services/storage/reactNativeKeychainStorage.ts

```diff
@@ -0,0 +1,91 @@
+import { logger } from "config/logger";
+import * as Keychain from "react-native-keychain";
+import { PersistentStorage } from "services/storage/storageFactory";
+
+/**
+ * Default service name used for the keychain
+ */
+const DEFAULT_SERVICE = "freighter_secure_storage";
+
+/**
+ * Sanitizes a key to be compatible with keychain
+ * Ensures that keys are valid for use with react-native-keychain
+ */
+function sanitizeKey(key: string): string {
```

Comment:
I'm a little concerned on sanitizing keys inside the getItem and setItem functions as it would be changing the original string input without the caller realizing it. Would it be safer (or less bug prone) to instead throw an error here and have the caller sanitize the key if they need to? What do you think?